### ISSUE TITLE
linux or docker instructions

### ISSUE THREAD

**Issue by AntouanK** [victorchall/EveryDream2trainer#10]
Hi there.

Will there be and doc/guide on how to install on linux or make a docker container for the trainer? ( ideally the second )

I tried myself but I get errors on installing the packages and I couldn't get it to run.

Thank you.

**Comment by victorchall** [comment]
Yeah the dependencies are a bit tricky, particularly xformers, I need to get runpod going then I can revisit linux/docker.

**Comment by chris-aeviator** [comment]
xformers, bitsandbytes an accellerate should all install fine via any official pytorch image. xformers requires you to build (via a pip command) if you don't use the conda repos, but it usally takes only 2 minutes.

**Comment by AntouanK** [comment]
could anyone attempt to make a docker image and if it works, share the Dockerfile here?

**Comment by chris-aeviator** [comment]
I‘m happy to share steps later for xf from a different docker file

**Comment by chris-aeviator** [comment]
Here's what I use for xformers in docker, lmk if it needs something else, removed many lines specific to my setup. I'm unsure if ED2 needs bitsandbytes (8bit adam). You might also not need huggingface transformers if the repo doesnt rely on the diffusers lib.

``
FROM nvidia/cuda:11.7.1-cudnn8-devel-ubuntu20.04
LABEL maintainer="Tofu AI"

ENV DEBIAN_FRONTEND=noninteractive

RUN apt update && \
    apt install -y bash \
                   build-essential \
                   git \
                   git-lfs \
                   curl \
                   ca-certificates \
                   python3.8 \
                   python3-pip \
                   python3.8-venv && \
    rm -rf /var/lib/apt/lists

# make sure to use venv (this might not be needed)
RUN python3 -m venv /opt/venv
ENV PATH="/opt/venv/bin:$PATH"

RUN python3 -m pip install --no-cache-dir --upgrade pip && \
    python3 -m pip install --no-cache-dir \
        torch \
        torchvision \
        torchaudio \
        --extra-index-url https://download.pytorch.org/whl/cu117 && \
    python3 -m pip install --no-cache-dir \
        accelerate \
        datasets \
        hf-doc-builder \
        huggingface-hub \
        modelcards \
        numpy \
        scipy \
        tensorboard \
        git+https://github.com/huggingface/transformers.git@main


RUN pip install ninja bitsandbytes
RUN pip install -v -U git+https://github.com/facebookresearch/xformers.git@main#egg=xformers

## install ED2 from here
``

**Comment by victorchall** [comment]
I use bits and bytes to support 8bit adam (significant performance improvement and ~1.5GB VRAM less) but that should be simple on linux, just install bitsandbytes 0.35.0, or newer versions may work as well on linux.

If bitsandbytes isn't installed, it will fall back to normal AdamW optimizer from torch, so its not strictly necessary.

**Comment by UglyStupidHonest** [comment]
I have it running fine with just installing the dependencies from I read in the bat file except the Windows stuff. The only tricky bit was that I had to update triton to get the xformers to work . (Ubuntu 20.4 lts)

**Comment by chris-aeviator** [comment]
things would be even easier with a requirements.txt file

**Comment by victorchall** [comment]
The windows_setup.cmd could be pretty easily stripped to requirements.txt, the patch_bnb is not required, and a different xformers install would be needed.  That's most if it.  Getting xformers to work in the biggest pain point.

**Comment by chris-aeviator** [comment]
what's the issue with xformers you face? It installed just fine with the official instructions every time I installed it, hayppy to assist

**Comment by victorchall** [comment]
Qslug has setup a container image here: 

https://ghcr.io/qslug/ed2runpod

Probably a few more hints here based on runpod/vast setup that could be useful for you:

https://github.com/victorchall/EveryDream2trainer/blob/main/doc/CLOUD_SETUP.md
